opensearchpy/connection/base.pyi

```diff
@@ -112,6 +112,8 @@ class Connection(object):
         response: Optional[str] = ...,
         exception: Optional[Exception] = ...,
     ) -> None: ...
-    def _raise_error(self, status_code: int, raw_data: str) -> NoReturn: ...
+    def _raise_error(
+        self, status_code: int, raw_data: str, content_type: Optional[str]
```

Comment:
Sorry, did not notice it, the correct type of `content_type` should be `str` since passing `None` will lead to an `AttributeError`. 
To be honest, I would prefer having `Optional`, because it is better than passing empty strings if there is no `Content-Type` available, but we need to have a check before trying to `.split(";")` in that case.

Reply:
Changed the implementation to be `Optional[str]` with default value of `None`

Reply:
Thanks!
Not sure if we want to parse JSON if there is no "Content-Type" header. Also `dict.get(value)` is sufficient (None is returned by default if key is missing).

Reply:
Now defaults to text/plain. removed default argument to `dict.get(k)`

Reply:
Thanks. Tests are [failing](https://github.com/opensearch-project/opensearch-py/runs/4605453962?check_suite_focus=true) with `E   AssertionError: no logs of level INFO or higher triggered on opensearch` - could you please take a look?

Reply:
Fixed. The test assumed content would be treated as application/json by default, which is no longer true.